src/compiler/parser.ts

```diff
@@ -3015,7 +3015,11 @@ namespace ts {
                 let newPrecedence = getBinaryOperatorPrecedence();
 
                 // Check the precedence to see if we should "take" this operator
-                if (newPrecedence <= precedence) {
+                if (token === SyntaxKind.AsteriskAsteriskToken && newPrecedence < precedence) {
+                    // ** operator is right-assocative
+                    break;
+                }
+                else if (token !== SyntaxKind.AsteriskAsteriskToken && newPrecedence <= precedence) {

```

Comment:
You can just keep this `if (newPrecedence <= precedence)`

Alternatively you could make an inner if:

```TypeScript
// Check the precedence to see if we can "take" the operator.
if (newPrecedence <= precedence) {
    // For left-associative operators, we can only accumulate left-heavy expressions
    // like this if the new precedence is less than or equal to than the current one.
    // However, for right-associative operators (i.e. really just **), the new operator's
    // precedence needs to be strictly less, since something like  a ** b ** c needs to be
    // parsed as the right-heavy expression a ** (b ** c).
    if (token !== SyntaxKind.AsteriskAsteriskToken || newPrecedence === precedence) {
        break;
    }
}